crates/storage/provider/src/traits/block.rs

```diff
@@ -71,17 +75,36 @@ pub trait BlockProvider:
         self.block(hash.into())
     }
 
-    /// Returns the block with matching tag from the database
+    /// Returns the block with matching number from database.
     ///
     /// Returns `None` if block is not found.
-    fn block_by_number_or_tag(&self, num: BlockNumberOrTag) -> Result<Option<Block>> {
+    fn block_by_number(&self, num: u64) -> Result<Option<Block>> {
         self.block(num.into())
     }
+}
 
-    /// Returns the block with matching number from database.
+/// Trait extension for `BlockProvider`, for types that implement `BlockId` conversion.
+#[auto_impl::auto_impl(&, Arc)]
+pub trait BlockProviderIdExt: BlockProvider + BlockIdProvider {
```

Comment:
okay so this bundles BlockProvider and BlockIdProvider together.

I guess we could also do

```rust
trait: BlockIdProvider: BlockProvider
```

and move the functions there, but this is kinda nice that it separates retrieving block numbers and block objects